src/buckets/components/createBucketForm/MeasurementSchemaUtils.ts

```diff
@@ -107,3 +107,81 @@ export const areSchemaUpdatesValid = schemaInfo => {
       !schemaInfo.hasUpdate || (schemaInfo.hasUpdate && schemaInfo.valid)
   )
 }
+
+// the MiniFileDnd component will catch any errors thrown here
+// and display them to the user; as this method is only called from within
+// the 'handleFileUpload' that is passed to and called by the MiniFileDnd Component.
+export const getColumnsFromFile = (contents: string, isCsv: boolean) => {
+  // do parsing here;  to check if in the correct format:
+  let columns = null
+  if (contents) {
+    if (isCsv) {
+      columns = csvToArray(contents)
+    } else {
+      // it's json:
+
+      // parse them; if proper/valid; great!  if not, set errors and do not proceed
+      // don't need to wrap this in try/catch since the caller of this function is inside a try/catch
+      columns = JSON.parse(contents)
+    }
+    if (!areColumnsProper(columns)) {
+      // set errors
+      throw {message: 'column file is not formatted correctly'}
+    }
+  }
+  return columns
+}
+/**
+ * take a csv file, as a string, and turn it into a javascript object
+ * based off: https://sebhastian.com/javascript-csv-to-array/
+ * if the value isn't present, then it isn't put into the object.
+ * all strings are trimmed as well.
+ */
+export const csvToArray = (contents: string, delimiter = ',') => {
+  // slice from start of text to the first \n index
+  // use split to create an array from string by delimiter
```

Comment:
all the comments in this method are repeating what the lines below them are doing - i think they should be removed because they're just noise.

https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/